src/devices/Ahtxx/AhtBase.cs

```diff
@@ -0,0 +1,159 @@
+        }
+
+        private enum Command : byte
+        {
+            Calibrate = 0xbe,
+            SoftRest = 0xba,
+            Measure = 0xac
+        }
+
+        private I2cDevice _i2cDevice = null;
+        private double _temperature;
+        private double _humidity;
+
+        /// <summary>
+        /// Initializes a new instance of the device connected through I2C interface.
+        /// </summary>
+        /// <paramref name="i2cDevice">Reference to the initialized I2C interface device</paramref>
+        public AhtBase(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+
+            SoftReset(); // even if not clearly stated in datasheet, start with a software reset to assure clear start conditions
+
+            // check whether the device indicates the need for a calibration cycle
+            // and perform calibration if indicated ==> c.f. datasheet, version 1.1, ch. 5.4
+            if (!IsCalibrated())
+            {
+                Calibrate();
+            }
+        }
+
+        /// <summary>
+        /// Gets the current temperature reading from the sensor.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Temperature GetTemperature()
+        {
+            Measure();
+            return new Temperature(_temperature, UnitsNet.Units.TemperatureUnit.DegreeCelsius);
+        }
+
+        /// <summary>
+        /// Gets the current humidity reading from the sensor.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Ratio GetHumidity()
+        {
+            Measure();
+            return new Ratio(_humidity, UnitsNet.Units.RatioUnit.Percent);
+        }
+
+        /// <summary>
+        /// Perform sequence to retrieve current readings from device
+        /// </summary>
+        private void Measure()
+        {
+            Span<byte> buffer = stackalloc byte[3]
+            {
+                (byte)Command.Measure,
+                0x33, // command parameters c.f. datasheet, version 1.1, ch. 5.4
+                0x00
+            };
+            _i2cDevice.Write(buffer);
+
+            while (IsBusy())
+            {
+                Thread.Sleep(10);
+            }
+
+            buffer = stackalloc byte[6];
+            _i2cDevice.Read(buffer);
+
+            Int32 rawHumidity = (buffer[1] << 12) | (buffer[2] << 4) | (buffer[3] >> 4);
+            _humidity = (rawHumidity * 100.0) / 0x100000;
+            Int32 rawTemperature = ((buffer[3] & 0xF) << 16) | (buffer[4] << 8) | buffer[5];
+            _temperature = ((rawTemperature * 200.0) / 0x100000) - 50;
+        }
+
+        /// <summary>
+        /// Perform soft reset command sequence
+        /// </summary>
+        private void SoftReset()
+        {
+            _i2cDevice.WriteByte((byte)Command.SoftRest);
+            Thread.Sleep(20); // reset requires 20ms at most, c.f. datasheet version 1.1, ch. 5.5
```

Comment:
Please cross check that the datasheet really says 20 Milliseconds. Most sensors have wait times in the order of nanoseconds only.

Reply:
Yes, it does, at least as a max. exec. time. However, let me try to figure out if this could be sped up using the busy flag. I'll built a prototype this weekend and check.